tests/test_main.py

```diff
@@ -1510,8 +1509,9 @@ class Entry(BaseModel):
     r.val = 101
     assert r.val == 101
     assert r.id == 1
-    with pytest.raises(TypeError, match='"id" has frozen set to True and cannot be assigned'):
+    with pytest.raises(ValidationError) as exc_info:
```

Comment:
It seems in v1, assigning to frozen fields raised a TypeError, rather than a ValidationError. Is this okay to change?

Reply:
Err, I guess it looks like frozen fields weren't even a thing in v1. But the xfailing test was using a TypeError. I'm assuming it's okay to make this a ValidationError.

Reply:
fine by me.